internal/retryhttp/client.go

```diff
@@ -133,3 +176,40 @@ func transport() http.RoundTripper {
 	}
 	return cleanhttp.DefaultPooledTransport()
 }
+
+// retryPolicy is a modified version of retryablehttp.DefaultRetryPolicy to handle
+// status codes differently.
+func retryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	// do not retry on context.Canceled or context.DeadlineExceeded
+	if ctx.Err() != nil {
+		return false, ctx.Err()
+	}
+
+	if err != nil {
+		if v, ok := err.(*url.Error); ok {
+			// Don't retry if the error was due to too many redirects.
+			if redirectsErrorRe.MatchString(v.Error()) {
+				return false, nil
+			}
+
+			// Don't retry if the error was due to an invalid protocol scheme.
+			if schemeErrorRe.MatchString(v.Error()) {
+				return false, nil
+			}
+
+			// Don't retry if the error was due to TLS cert verification failure.
+			if _, ok := v.Err.(x509.UnknownAuthorityError); ok {
+				return false, nil
+			}
+		}
+
+		// The error is likely recoverable so retry.
+		return true, nil
+	}
+
+	return isRetryableStatus(resp.StatusCode), nil
```

Comment:
Wouldn't err be non-nil for status codes over the 200 series? Meaning this wouldn't hit due to the err check above.

Reply:
I think you're right... if that's the case then I don't know if the default retry client was ever retrying on 500 errors 😕 

https://github.com/ActiveState/cli/blob/DX-1723/vendor/github.com/hashicorp/go-retryablehttp/client.go#L440

Reply:
No I'm wrong, seems this should be ok:

![image](https://user-images.githubusercontent.com/278649/235209499-2bcd27cf-234b-4c1d-835f-2cbe26c0b4b8.png)